src/cca/scripts/background.ts

```diff
@@ -0,0 +1,41 @@
+/// <reference path='../../../third_party/cordova/splashscreen.d.ts'/>
+
+/**
+ * background.ts
+ *
+ * This file must be included *after* the freedom script and manifest are
+ * loaded.
+ */
+
+import { CordovaBrowserApi } from './cordova_browser_api';
+
+import * as uproxy_core_api from '../../interfaces/uproxy_core_api';
+
+declare const freedom: freedom.FreedomInCoreEnv;
+
+export interface OnEmitModule extends freedom.OnAndEmit<any,any> {};
+export interface OnEmitModuleFactory extends
+  freedom.FreedomModuleFactoryManager<OnEmitModule> {};
+
+export var moduleName = 'uProxy App Top Level';
+
+// TODO(fortuna): Move browser api logic directly here. No need for the extra layer.
+var browserApi :CordovaBrowserApi = new CordovaBrowserApi();
+
+console.log('Loading core');
+export var uProxyAppChannel = freedom(
```

Comment:
This is used by `CordovaCoreConnector`:

    <any>bgPage).ui_context.uProxyAppChannel